From 2511d56993d8c7412645bd17d45645cd95d5ead8 Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Mon, 29 Nov 2010 14:46:56 +0100
Subject: [PATCH] spi: Clear dma_mask for xilinx spi controller

mmc_spi driver tests if dma is available.
Microblaze systems supports DMA but xilinx_spi
controller doesn't support DMA transfers that's
why it is necessary to do this workaround to
clear dma_mask in spi controller.

mmc_spi then checks spi->master->dev.parent->dma_mask
which is cleared and dma mode is not used.

Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 drivers/spi/spi-xilinx.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/spi/spi-xilinx.c b/drivers/spi/spi-xilinx.c
index 28444fd..78774e9 100644
--- a/drivers/spi/spi-xilinx.c
+++ b/drivers/spi/spi-xilinx.c
@@ -365,6 +365,9 @@ struct spi_master *xilinx_spi_init(struct device *dev, struct resource *mem,
 	if (!master)
 		return NULL;
 
+	/* clear the dma_mask, to try to disable use of dma */
+	master->dev.dma_mask = 0;
+
 	/* the spi->mode bits understood by this driver: */
 	master->mode_bits = SPI_CPOL | SPI_CPHA;
 
-- 
1.7.0.4

